src/main/objects/soprano/SopranoSolver2.js

```diff
@@ -0,0 +1,96 @@
+.import "../soprano/HarmonicFunctionGenerator.js" as HarmonicFunctionGenerator
+.import "../soprano/SopranoRulesChecker.js" as SopranoRulesChecker
+.import "../algorithms/SopranoGraphBuilder.js" as SopranoGraphBuilder
+.import "../harmonic/Exercise.js" as HarmonicFunctionsExercise
+.import "../utils/Utils.js" as Utils
+.import "../algorithms/Dikstra.js" as Dikstra
+.import "../harmonic/Exercise.js" as Exercise
+.import "../harmonic/Solver2.js" as HarmonisationSolver
+.import "../commons/ExerciseSolution.js" as ExerciseSolution
+.import "../commons/Errors.js" as Errors
+.import "../harmonic/ChordGenerator.js" as ChordGenerator
+.import "../harmonic/ChordRulesChecker.js" as RulesChecker
+.import "../algorithms/Graph.js" as Graph
+.import "../algorithms/GraphBuilder.js" as GraphBuilder
+
+function SopranoSolver(exercise, punishmentRatios){
+
+    this.exercise = exercise;
+    this.harmonicFunctionGenerator = new HarmonicFunctionGenerator.HarmonicFunctionGenerator(this.exercise.possibleFunctionsList, this.exercise.key, this.exercise.mode);
+    this.numberOfRetry = 10;
+    this.punishmentRatios = punishmentRatios;
+    this.sopranoRulesChecker = new SopranoRulesChecker.SopranoRulesChecker(this.exercise.key, this.exercise.mode, this.punishmentRatios);
+
+    this.mapToHarmonisationExercise = function(harmonicFunctions){
+        var measures = []
+        var current_measure = []
+        var counter = 0
+        var tmp = "";
```

Comment:
To bym z if (DEBUG) zrobił

Reply:
Podobnie niżej, gdzie to tmp jest

Reply:
W zasadzie niepotrzebnie to zacommitowałem, usuwam gdyż raczej nie przyda się w przyszłości